polkadot/xcm/xcm-builder/src/universal_exports.rs

```diff
@@ -438,16 +441,29 @@ impl<Bridge: HaulBlob, BridgedNetwork: Get<NetworkId>, Price: Get<MultiAssets>>
 		message: &mut Option<Xcm<()>>,
 	) -> Result<((Vec<u8>, XcmHash), MultiAssets), SendError> {
 		let bridged_network = BridgedNetwork::get();
-		ensure!(&network == &bridged_network, SendError::NotApplicable);
+		ensure!(&network == &bridged_network.0, SendError::NotApplicable);
 		// We don't/can't use the `channel` for this adapter.
 		let dest = destination.take().ok_or(SendError::MissingArgument)?;
-		let universal_dest = match dest.pushed_front_with(GlobalConsensus(bridged_network)) {
-			Ok(d) => d.into(),
-			Err((dest, _)) => {
-				*destination = Some(dest);
-				return Err(SendError::NotApplicable)
-			},
-		};
+
+		// Let's resolve the known/supported XCM version for the destination because we don't know
+		// if it supports the same/latest version.
+		let (universal_dest, version) =
+			match dest.pushed_front_with(GlobalConsensus(bridged_network.0)) {
+				Ok(d) => {
+					let version = Bridge::check_version_for(
```

Comment:
Do we need to do this ? In theory we already checked if XCM version on the destination is ok inside the bridge hub router (in the `SendXcm` implementation).

Reply:
yes, we need, please, check description: `Incompatibility when BHW uses a different version than BHR...`,
there are two validations:
1. between AssetHubs - if XCM program is compatible (yes, it is not 100% because of HaulBlobExporter - see my [comment here](https://github.com/paritytech/polkadot-sdk/pull/2481#discussion_r1417083953)) 
2. between BridgeHubs

and here in `HaulBlobExporter` we try to ensure that XCM program between BridgeHubs is compatible.

So in practice, imagine we deployed: AHK (v3) -> BHK(v3) -> BHP(v3) -> AHP(v3).
Kusama is enacted sooner with XCMv4: AHK (v4) -> BHK(v4) -> BHP(v3) -> AHP(v3).
So now you see, that `BHK(v4) -> BHP(v3)`  HaulBlobExporter on BHK can not use `VersionedXcm` with latest XCMv4 because it will fail on BHP, so this PR fixes this, we will keep xcm version for BHP on BHK in `pallet_xcm::SupportedVersion`, so even if BHK is on XCMv4 the `HaulBlobExporter` will encode message to the XCMV3 (if compatible). It is easier to handle dropped messages on one side of consensus than across and waste resources.